### ISSUE TITLE
BraintreeUIKit alternate.

### ISSUE THREAD

**Issue by vadivelmuniswamy** [braintree/braintree-ios-drop-in#348]
Our product has customized fields to collect the card details.
But it's deprecated from 9.0.0
What is the alternate way to use them?
Is there any possibilities to release them again?

**Comment by scannillo** [comment]
In v9.0.0 we removed the deprecated `BraintreeUIKit` framework (see [CHANGELOG](https://github.com/braintree/braintree-ios-drop-in/blob/master/CHANGELOG.md#900-beta1-2021-03-29)).

We instead encourage you to use the [full drop-in UI](https://developer.paypal.com/braintree/docs/start/hello-client/ios/v5#present-drop-in-ui), which does contain and handle card form presentation (and card detail collection).

**Comment by fredadda-fiverr** [comment]
@scannillo hi, does that mean that fields such as BTUIKFormField are no longer available if we switch to SPM (which requires v9)?
We need to use a complete card form presentation?

**Comment by scannillo** [comment]
👋  @fredadda-fiverr - No matter which package manager you use for v9, `BraintreeUIKit` has been removed. So for v9 you would have to use the full Drop-In UI. If you want to to customize your components and card fields, we encourage you use `braintree_ios` instead.

However, I do encourage you to comment here with what type of UI components you would like the SDK to have, so we can gauge a sense of interest from other devs as well.

**Comment by fredadda-fiverr** [comment]
Thanks for the answer! 
Here at Fiverr we are mostly interested in a screen to add Credit Card details (C/C number, First name / last name, expiration date and CVV), with the possibility to add a few custom fields around these C/C details. We wanted to apply our own design to this screen, that's why we used only individuals fields from the BTUIKit lib.

However, we used an old version of the lib (Braintree 4.36.0), and a quite old design too, so maybe it's the occasion to revamp it a bit 😄 
I will check what we can do with the full Drop-in UI, if we can tailor it for our needs. If not, I will check Braintree.
Thanks!

**Comment by fredadda-fiverr** [comment]
Hello @scannillo!
I tried to implement the BTDropInController in our workflow, but I realize it's a complete tokenization solution of the Credit Card.
What we need are just the individual fields with their logic (for example the C/C entry field with the Luhn algorithm, expiration date field with month / selection, etc.)
![IMG_9637D7A40F84-1](https://user-images.githubusercontent.com/62248743/130084112-61ddf115-e1a8-484b-8c5d-a6aa3d70b33f.jpeg)

Do you think that's something that exists in the `braintree_ios` lib, or do we have to recreate the fields from scratch?
Thanks for your help

**Comment by fredadda-fiverr** [comment]
@scannillo Hi again
I didn't see any possibility to customize UI components in Braintree iOS. Did you mean to create our own UI components (textFields and all) and use Braintree for tokenization?

**Comment by scannillo** [comment]
👋  @fredadda-fiverr, so we only really offer the DropIn UI with the full tokenization attached, though the fonts & color scheme are customizable. We currently don't offer any standalone card fields.

**Comment by fredadda-fiverr** [comment]
@scannillo Hi! Just another question: if we then decide to build our own UI, is there a validation logic I can use from Braintree or Braintree-dropIn, like the Luhn algorithm for C/C, or are there other standalone logic tools that are exposed?
Thanks

**Comment by scannillo** [comment]
@fredadda-fiverr No, the `braintree_ios` doesn't include any Luhn verification. You can check out our[ Card docs](https://developer.paypal.com/braintree/docs/guides/credit-cards/overview) to read more about the features offered.

**Comment by timonchev** [comment]
I'm in same situation, can we use somehow UI textfields with Braintree verification?

**Comment by jaxdesmarais** [comment]
The Drop-in SDK is now in an [inactive status](https://developer.paypal.com/braintree/docs/guides/client-sdk/deprecation-policy/ios/v6/#status-categories). Starting September 1, 2026 the Drop-in SDK will move to a [deprecated status](https://developer.paypal.com/braintree/docs/guides/client-sdk/deprecation-policy/ios/v6/#status-categories) and we will no longer make any updates to this SDK. Processing will be supported for 1 year after the deprecation date, but you should upgrade immediately to avoid any disruption.

Starting September 1, 2027 the Drop-in SDK will move to an [unsupported status](https://developer.paypal.com/braintree/docs/guides/client-sdk/deprecation-policy/ios/v6/#status-categories) and will no longer supported by Braintree developers or Braintree Support. Processing for unsupported SDKs can be suspended at any time.

Please migrate to the [Braintree SDK](https://github.com/braintree/braintree_ios) to continue processing and receiving updates. We currently support native Venmo and PayPal buttons on the core SDK with card fields coming soon.
